cli/permissions.rs

```diff
@@ -435,4 +541,171 @@ mod tests {
+      PermissionAccessorState::Allow
+    );
+
+    let perms1 = DenoPermissions::from_flags(&DenoFlags {
+      write_whitelist: whitelist.clone(),
+      ..Default::default()
+    });
+    set_prompt_result(true);
+    assert_eq!(
+      perms1.request_write(&Some("/foo/baz")),
+      PermissionAccessorState::Allow
+    );
+
+    let perms2 = DenoPermissions::from_flags(&DenoFlags {
+      write_whitelist: whitelist.clone(),
+      ..Default::default()
+    });
+    set_prompt_result(false);
+    assert_eq!(
+      perms2.request_write(&Some("/foo/baz")),
+      PermissionAccessorState::Deny
+    );
+  }
+
+  #[test]
+  fn test_permission_request_net() {
+    let whitelist = svec!["localhost:8080"];
+
+    let perms0 = DenoPermissions::from_flags(&DenoFlags {
+      net_whitelist: whitelist.clone(),
+      ..Default::default()
+    });
+    set_prompt_result(false);
+    // If the url matches the whitelist item, then the result is `allow`
+    // regardless of prompt result
+    assert_eq!(
+      perms0
+        .request_net(&Some("http://localhost:8080/"))
+        .expect("Testing expect"),
+      PermissionAccessorState::Allow
+    );
+
+    let perms1 = DenoPermissions::from_flags(&DenoFlags {
+      net_whitelist: whitelist.clone(),
+      ..Default::default()
+    });
+    set_prompt_result(true);
+    assert_eq!(
+      perms1
+        .request_net(&Some("http://deno.land/"))
+        .expect("Testing expect"),
+      PermissionAccessorState::Allow
+    );
+
+    let perms2 = DenoPermissions::from_flags(&DenoFlags {
+      net_whitelist: whitelist.clone(),
+      ..Default::default()
+    });
+    set_prompt_result(false);
+    assert_eq!(
+      perms2
+        .request_net(&Some("http://deno.land/"))
+        .expect("Testing expect"),
+      PermissionAccessorState::Deny
+    );
+
+    let perms3 = DenoPermissions::from_flags(&DenoFlags {
+      net_whitelist: whitelist.clone(),
+      ..Default::default()
+    });
+    set_prompt_result(true);
+    assert!(perms3.request_net(&Some(":")).is_err());
+  }
+
+  #[test]
+  fn test_permissions_request_env() {
+    let perms0 = DenoPermissions::from_flags(&DenoFlags {
+      ..Default::default()
+    });
+    set_prompt_result(true);
+    assert_eq!(perms0.request_env(), PermissionAccessorState::Allow);
+
+    let perms1 = DenoPermissions::from_flags(&DenoFlags {
+      ..Default::default()
+    });
+    set_prompt_result(false);
+    assert_eq!(perms1.request_env(), PermissionAccessorState::Deny);
+  }
+
+  #[test]
+  fn test_permissions_request_hrtime() {
+    let perms0 = DenoPermissions::from_flags(&DenoFlags {
+      ..Default::default()
+    });
+    set_prompt_result(true);
+    assert_eq!(perms0.request_hrtime(), PermissionAccessorState::Allow);
+
+    let perms1 = DenoPermissions::from_flags(&DenoFlags {
+      ..Default::default()
+    });
+    set_prompt_result(false);
+    assert_eq!(perms1.request_hrtime(), PermissionAccessorState::Deny);
+  }
```

Comment:
Sorry - I didn't see these tests when I wrote my previous comment. 

But these still don't seem to be using openpty... I need to look into it more.

Reply:
These tests replace `prompt_permissions` with stub function and its returning values are controlled by `set_prompt_result` calls.